R/expect-no-condition.R

```diff
@@ -93,13 +93,18 @@ expect_no_ <- function(base_class,
   check_dots_used(action = warn, call = error_call)
   matcher <- cnd_matcher(class %||% base_class, regexp, ...)
 
+  ignore_deprec <- base_class == "warning" && is.null(regexp) && is.null(class)
+
   capture <- function(code) {
     try_fetch(
       code,
       !!base_class := function(cnd) {
         if (!matcher(cnd)) {
           return(zap())
         }
+        if (!ignore_deprec && is_deprecation(cnd)) {
```

Comment:
Note: I find it hard to follow the logic here and how it interacts with the one in `cnd_matcher()`.

Reply:
Oops, because I accidentally implemented it twice.